src/gausskernel/storage/access/k2/pg_gate_api.cpp

```diff
@@ -584,12 +579,40 @@ K2PgStatus PgGate_ExecInsert(K2PgOid database_oid,
                              K2PgOid table_oid,
                              bool upsert,
                              bool increment_catalog,
-                             const std::vector<K2PgAttributeDef>& columns) {
+                             std::vector<K2PgAttributeDef>& columns) {
     elog(LOG, "PgGateAPI: PgGate_ExecInsert %d, %d", database_oid, table_oid);
     auto catalog = pg_gate->GetCatalogClient();
 
     skv::http::dto::SKVRecord record;
-    K2PgStatus status = makeSKVRecordFromK2PgAttributes(database_oid, table_oid, columns, record);
+    std::shared_ptr<k2pg::PgTableDesc> pg_table = k2pg::pg_session->LoadTable(database_oid, table_oid);
+    // check if the table has k2pgrowid system column
+    if (pg_table->FindColumn(k2pg::to_underlying(k2pg::PgSystemAttrNum::kPgRowId)) != NULL) {
+        // check if k2pgrowid has already been included in passed in columns
+        bool kPgRowIdProvided = false;
+        for (const auto& column: columns) {
+            if (column.attr_num == k2pg::to_underlying(k2pg::PgSystemAttrNum::kPgRowId)) {
+                kPgRowIdProvided = true;
+                break;
+            }
+        }
+        if (!kPgRowIdProvided) {
+            // generate a row_id to populate the kPgRowId column
+            std::string row_id = k2pg::pg_session->GenerateNewRowid();
```

Comment:
You could have GenerateNewRowid return a Datum directly but it is up to you

Reply:
Datum is a complex data type in postgres, it isn't easy to check the value as std:string. Prefer to keep std:string here. Apart from that, tried to add Datum here, but the header files messed up easily. As a result, keep it as it is.